src/sage/categories/rings.py

```diff
@@ -1943,7 +1943,7 @@ def is_unit(self):
                     sage: e.is_unit()
                     False
                 """
-                return all([_.is_unit() for _ in self.cartesian_factors()])
+                return all([x.is_unit() for x in self.cartesian_factors()])
```

Comment:
no need for the brackets inside all : use `all(x.is_unit() for x in self.cartesian_factors())`